docs/environment-setup.md

```diff
@@ -117,3 +117,28 @@ Docker Desktop for Windows is not available on Home Edition, and you cannot run
   1. Download Windows 10 Education Edition from the [Seneca Software Center](https://senecacollege.onthehub.com/WebStore/OfferingDetails.aspx?o=c0bd2c36-a530-e511-940e-b8ca3a5db7a1)
   1. Update your OS using the installation instructions.
   1. Use [Windows 10 Education Edition](#Windows-10-Pro,-Enterprise,-or-Education) set up instructions.
+
+## After installing the prerequisites:
+
+### SAML Setup
+
+1. Run `./tools/generate_ssl_certs.sh` in terminal, this will create a `certs` directory and two pem files `key.pem` and `cert.pem` inside. On Windows, only `key.pem` will be created in /certs.
+2. You will also need to create a `idp_key.pem` file in the certs folder, copy and paste the following key into `idp_key.pem`:
+
+`MIIDXTCCAkWgAwIBAgIJALmVVuDWu4NYMA0GCSqGSIb3DQEBCwUAMEUxCzAJBgNVBAYTAkFVMRMwEQYDVQQIDApTb21lLVN0YXRlMSEwHwYDVQQKDBhJbnRlcm5ldCBXaWRnaXRzIFB0eSBMdGQwHhcNMTYxMjMxMTQzNDQ3WhcNNDgwNjI1MTQzNDQ3WjBFMQswCQYDVQQGEwJBVTETMBEGA1UECAwKU29tZS1TdGF0ZTEhMB8GA1UECgwYSW50ZXJuZXQgV2lkZ2l0cyBQdHkgTHRkMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAzUCFozgNb1h1M0jzNRSCjhOBnR`
+
+3. Create a file called `.env` in the root directory. Copy and paste everything in `env.example` (also located in root) into your newly created `.env`. Ensure all related SAML2 information is filled out.
+
+- SAML2_BASE_URI
+- SAML2_CLIENT_ID
+- SAML2_CLIENT_SECRET
+- SAML2_REDIRECT_URI
+
+_Note: The `env.example` file has examples commented on top of each variable_
+
+### Now you can start Redis without errors using one of the following methods:
+
+- Using Docker `docker run -d -p 6379:6379 redis`
```

Comment:
I think starting `redis` using `docker-compose up redis` might be a bit better since it uses our settings in our `docker-compose.yml` file (hopefully #574 will get fixed soon).

Also, the option `-d` means `detach` which I don't think it's ideal in this case since it returns control to the user in the command line after deployment and we would have to provide another command to stop `redis`. Deploying without `-d` forces you to use `Ctrl+c` to stop `redis`, which I think it's easier to remember than a second command for stopping it.

You can test both options and see which one would be easier to remember.

Reply:
Ah okay so I got `sudo docker-compose up redis` to work but when I run `npm start`, I get an error saying `Redis is not running on your machine`. Am I suppose to run `npm start` in the docker container or something? I just started using Docker recently...

Reply:
haha, used `Ctrl+C` to stop redis and got this cute little message:
```
Gracefully stopping... (press Ctrl+C again to force)
Stopping telescope_redis_1 ... 
```
I definitely prefer this way more.
I still don't know how to run `npm start` though

Reply:
Check your `.en` file, and make sure `REDIS_PORT` is `6379`. If it's` 6739` it's because #574. Awesome that now we have that error message for this situation, makes things sooo much easier.

Reply:
oh @cindyledev also, I think it's better if we don't use `sudo` with `docker-compose`. Let me know if it works for you without `sudo` (and using port `6379`).

Reply:
And, something else, let's throw `--build` in the mix, Sorry that I'm doing this in so many messages.
The idea is that whenever you launch `docker-compose` you want to rebuild the container in case changes have been made. For that, we add `--build`.
`docker-compose up --build redis`.

Reply:
So I changed REDIS_PORT to 6379 and ran
`sudo systemctl start docker`
`sudo docker-compose up --build redis`
I can't run either of those without `sudo` for some reason.
I opened a second terminal and ran `npm start` but this is what I got in `localhost:3000`
![VirtualBox_Fedora_26_01_2020_14_44_35](https://user-images.githubusercontent.com/36822198/73140757-6ab18d80-404a-11ea-8d4a-2b42c15a5927.png)
Did I do something wrong?

I get the same result in `localhost:3000` when I run `redis-server` too, it didn't look like this before

Reply:
For [systemctl](https://www.linode.com/docs/quick-answers/linux-essentials/introduction-to-systemctl/) you need `sudo`, that's ok.

I'm not sure why it doesn't let you use `docker-compose` without `sudo`, did you add your user to the `docker` group? try `groups $USER` in a terminal and check if docker is in the list of groups. If not, add it with `usermod -aG docker $USER` and reboot.
Note: Instead of `$USER` you can use your user, `USER` is just a variable that has the name of the user currently logged in.

I think what you see in `localhost:3000` is fine, that's what Telescope looks like right now after merging #531 (check http://dev.telescope.cdot.systems).

Reply:
We lost a bunch of CSS and other assets with the Gatsby code being merged, so this is a temporary problem.  I fixed the planet due to the same.
<img width="749" alt="Screen Shot 2020-01-26 at 3 03 26 PM" src="https://user-images.githubusercontent.com/427398/73140994-1d82eb00-404d-11ea-9237-b66b1dfb85d1.png">

Reply:
I ran `usermod -aG docker $USER` and rebooted, now I'm able to run `docker-compose` without `sudo`. Thanks a lot for all this help :)